src/gentropy/eqtl_catalogue.py

```diff
@@ -1,62 +1,79 @@
-"""Step to run eQTL Catalogue study table ingestion."""
+"""Step to run eQTL Catalogue credible set and study index ingestion."""
 
 from __future__ import annotations
 
+import pandas as pd
+import pyspark.sql.functions as f
+
 from gentropy.common.session import Session
+from gentropy.datasource.eqtl_catalogue.finemapping import EqtlCatalogueFinemapping
 from gentropy.datasource.eqtl_catalogue.study_index import EqtlCatalogueStudyIndex
-from gentropy.datasource.eqtl_catalogue.summary_stats import EqtlCatalogueSummaryStats
 
 
 class EqtlCatalogueStep:
-    """eQTL Catalogue ingestion step."""
+    """eQTL Catalogue ingestion step.
+
+    From SuSIE fine mapping results (available at [their FTP](https://ftp.ebi.ac.uk/pub/databases/spot/eQTL/susie/) ), we extract credible sets and study index datasets from gene expression QTL studies.
+    """
 
     def __init__(
         self,
         session: Session,
+        mqtl_quantification_methods: list[str],
         eqtl_catalogue_paths_imported: str,
         eqtl_catalogue_study_index_out: str,
-        eqtl_catalogue_summary_stats_out: str,
+        eqtl_catalogue_credible_sets_out: str,
     ) -> None:
         """Run eQTL Catalogue ingestion step.
 
         Args:
             session (Session): Session object.
-            eqtl_catalogue_paths_imported (str): Input eQTL Catalogue study index path.
+            mqtl_quantification_methods (list[str]): Molecular trait quantification methods to ingest. Available options in https://github.com/eQTL-Catalogue/eQTL-Catalogue-resources/blob/master/data_tables/dataset_metadata.tsv
+            eqtl_catalogue_paths_imported (str): Input eQTL Catalogue fine mapping results path.
             eqtl_catalogue_study_index_out (str): Output eQTL Catalogue study index path.
-            eqtl_catalogue_summary_stats_out (str): Output eQTL Catalogue summary stats path.
+            eqtl_catalogue_credible_sets_out (str): Output eQTL Catalogue credible sets path.
         """
-        # Fetch study index.
-        df = session.spark.read.option("delimiter", "\t").csv(
-            eqtl_catalogue_paths_imported, header=True
+        # Extract
+        pd.DataFrame.iteritems = pd.DataFrame.items
+        studies_metadata = session.spark.createDataFrame(
+            pd.read_csv(EqtlCatalogueStudyIndex.raw_studies_metadata_path, sep="\t"),
+            schema=EqtlCatalogueStudyIndex.raw_studies_metadata_schema,
+        ).filter(f.col("quant_method").isin(list(mqtl_quantification_methods)))
```

Comment:
to include processing inside the datasource (requires passing the `Session` to the datasource)